rfcs/0001-connection-pooling.md

```diff
@@ -0,0 +1,71 @@
+---
+feature: Connection Pooling
+start-date: 2021-02-04
+author: kiwicopple
+co-authors: steve-chavez, dragarcia
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+We would like to explore connection pooling on Supabase. This RFC is intended to decide:
+
+- Whether we should provide a pooler
+- Which connection pooler we should use
+- Where in the stack it would be installed - i.e. if should bundle it with the Postgres build
+
+
+# Motivation
+[motivation]: #motivation
+
+In Postgres, every connection is a process. Because of this, a lot of connections to the database can be very expensive on memory. 
+
+When connecting to Postgres database from serverless functions, there is no connection pooling, and so the server needs to maintain hundreds/thousands of connections.
```

Comment:
Transaction pooling and statement pooling(see [pgbouncer types of pooling](https://www.pgbouncer.org/features.html)) might help there. The disadvantage of those is that they cannot use prepared statements for example, so we'd need to check for support on common client libs(too much work, maybe just document it).

Seems DO also recommends transaction pooling: https://assets.digitalocean.com/articles/managed_db_pools/connection_pool_blank.png

Reply:
Agreed on this - when I look at the additional tooling around Postgres, pgbouncer seems to be the one supported most frequently (eg, `pgmetrics`, various postgres `operators`).